### PR TITLE
[FEATURE] Geocoding wizard

### PR DESCRIPTION
**Pull request by georgringer**
This PR adds a leaflet/openstreetmap based mapwizard with geocoding, aiming to:
- not force user to manually add coordinates
- saving as much geocoding calls as possible, so geocoding is only done on first map-opening and only if coordinate-fields are empty.

### CODE REVIEW

File: Resources/Public/JavaScript/LeafletBackend.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        LeafBE.$fieldLat = LeafBE.$element.attr('data-namelat');
+        LeafBE.$fieldLon = LeafBE.$element.attr('data-namelon');
+        LeafBE.$fieldLatActive = LeafBE.$element.attr('data-namelat-active');
+
+        // add the container to display the map as a nice overlay
+        $('body').append(
+            '<div id="t3js-location-map-wrap">' +
+            '<div class="t3js-location-map-title">' +
+            '<div class="btn-group"><a href="#" class="btn btn-default" title="Close" id="t3js-ttaddress-close-map">' +
+            '<img src="/typo3/sysext/core/Resources/Public/Icons/T3Icons/actions/actions-close.svg" width="16" height="16"></a>' +
```
Comment (georgringer) [inline Resources/Public/JavaScript/LeafletBackend.js]: don't hardcode that, won't also work in sub directory

File: Classes/FormEngine/FieldControl/LocationMapWizard.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+/**
+ * This file is part of the "tt_address" Extension for TYPO3 CMS.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE.txt file that was distributed with this source code.
+ */
+use TYPO3\CMS\Backend\Form\AbstractNode;
+
+class LocationMapWizard extends AbstractNode
```
Comment (georgringer) [inline Classes/FormEngine/FieldControl/LocationMapWizard.php]: wanna add a class comment?

File: Classes/FormEngine/FieldControl/LocationMapWizard.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        $resultArray['linkAttributes']['data-geocodeurl'] = $geoCodeUrl;
+        $resultArray['linkAttributes']['data-geocodeurlshort'] = $geoCodeUrlShort;
+        $resultArray['linkAttributes']['data-namelat'] = htmlspecialchars($nameLat);
+        $resultArray['linkAttributes']['data-namelon'] = htmlspecialchars($nameLon);
+        $resultArray['linkAttributes']['data-namelat-active'] = htmlspecialchars($nameLatActive);
+        $resultArray['linkAttributes']['data-tiles'] = htmlspecialchars($tilesUrl);
+        $resultArray['linkAttributes']['data-copy'] = $tilesAttribution;
+        $resultArray['stylesheetFiles'][] = 'EXT:tt_address/Resources/Public/Css/leaflet.css';
+        $resultArray['stylesheetFiles'][] = 'EXT:tt_address/Resources/Public/Css/leafletBackend.css';
+        $resultArray['requireJsModules'][] = 'TYPO3/CMS/TtAddress/leaflet-1.4.0';
```
Comment (georgringer) [inline Classes/FormEngine/FieldControl/LocationMapWizard.php]: is this really needed?

Comment (BastianBalthasarBux) [reply Classes/FormEngine/FieldControl/LocationMapWizard.php]: Which line do you mean exactly?

Comment (BastianBalthasarBux) [reply Classes/FormEngine/FieldControl/LocationMapWizard.php]: Yes, we need that, as discussed on tel.

File: Classes/FormEngine/FieldControl/LocationMapWizard.php (hunk 2 above)
Comment (BastianBalthasarBux) [reply Classes/FormEngine/FieldControl/LocationMapWizard.php]: Done.

File: Resources/Public/JavaScript/LeafletBackend.js (hunk 1 above)
Comment (BastianBalthasarBux) [reply Resources/Public/JavaScript/LeafletBackend.js]: Done.

**Pull request closed without merging**